### PR TITLE
QW-2273: Add the ability to disable ANSI terminal codes in logging.

### PR DESCRIPTION
**Pull request by ChillFish8**
This adds the ability to pass a `--no-ansi-logging` flag or `QW_NO_ANSI_LOGGING` environment variable in order to disable tracing_subscriber from injecting ANSI terminal codes, reducing the noise when logging to things like files which do not support rendering the codes.

### CODE REVIEW

**Comment by fmassot** [comment]
@ChillFish8 I think we can follow this standard, it's pretty standard:  https://no-color.org/

**Comment by ChillFish8** [comment]
> @ChillFish8 I think we can follow this standard, it's pretty standard:  https://no-color.org/
> 
> 

That's true, I was unsure whether it's a good idea to call it `no color` or not as it affects all ANSI codes rather than just the colours, which could be miss leading.

But if everyone is using the same naming then it's probably worth changing.

**Comment by fmassot** [comment]
mm, good point, so maybe `NO_ANSI` is better here.

**Comment by ChillFish8** [comment]
I think we should go with  `NO_COLOR` for familiarity, I think if people want to disable ANSI codes it's probably because of the colours and the noise they bring. Probably better to be consistent with what others are doing.

**Comment by fmassot** [comment]
I like the `NO_COLOR` too, let's go for this one :)

File: quickwit/quickwit-cli/src/cli.rs [hunk 1]
```diff
@@ -38,6 +38,18 @@ pub fn build_cli<'a>() -> Command<'a> {
                 .global(true)
                 .display_order(1),
         )
+        .arg(
+            Arg::new("no-ansi")
```
Comment (fmassot) [inline quickwit/quickwit-cli/src/cli.rs:42]:
```suggestion
            Arg::new("no-color")
```

File: quickwit/quickwit-cli/src/main.rs [hunk 2]
```diff
@@ -130,7 +135,11 @@ async fn main() -> anyhow::Result<()> {
     #[cfg(feature = "jemalloc")]
     start_jemalloc_metrics_loop();
 
-    setup_logging_and_tracing(command.default_log_level(), &build_info)?;
+    setup_logging_and_tracing(
+        command.default_log_level(),
+        !matches.is_present("no-ansi"),
```
Comment (fmassot) [inline quickwit/quickwit-cli/src/main.rs:140]:
```suggestion
        !matches.is_present("no-clor"),
```

Comment (ChillFish8) [reply quickwit/quickwit-cli/src/main.rs]: Have changed this, thought it would make sense to keep it as ansi but it ended up looking more confusing 😅

**Review by fmassot — APPROVED** [review]

**Pull request merged**